base/string.jl

```diff
@@ -441,10 +441,11 @@ function rsearchindex(s::ByteString, t::ByteString)
 end
 
 function rsearchindex(s::ByteString, t::ByteString, i::Integer)
-    if length(t) == 1
+    l = length(t)
```

Comment:
I don't like that we call `length`, which is an O(n) operation, just to check whether the length is 0 or 1.   It would be more efficient to call `isempty` to check if it is length zero, and `nextind(t, 1) > endof(t)` to check whether it is length 1.